build-system/tasks/runtime-test/status-report.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const argv = require('minimist')(process.argv.slice(2));
+const log = require('fancy-log');
+const requestPromise = require('request-promise');
+const {cyan, green, yellow} = require('ansi-colors');
+const {gitCommitHash} = require('../../git');
+const {isTravisPullRequestBuild} = require('../../travis');
+
+const reportBaseUrl = 'https://amp-test-status-bot.appspot.com/v0/tests';
+
+const IS_INTEGRATION = !!argv.integration;
+const IS_LOCAL_CHANGES = !!argv['local-changes'];
+const IS_SAUCELABS = !!(argv.saucelabs || argv.saucelabs_lite);
+const IS_SINGLE_PASS = !!argv.single_pass;
+const IS_UNIT = !!argv.unit;
+
+function inferTestType() {
+  if (IS_INTEGRATION) {
+    if (IS_SAUCELABS) {
+      // TODO(danielrozenberg): report integration on saucelabs
+      return null;
+    }
+    return 'integration' + (IS_SINGLE_PASS ? '-single-pass' : '');
+  } else if (IS_LOCAL_CHANGES) {
+    return 'local-changes';
+  } else if (IS_UNIT) {
+    return 'unit' + (IS_SAUCELABS ? '-saucelabs' : '');
```

Comment:
From looking at https://github.com/ampproject/amphtml/pull/21407/checks?check_run_id=81073337,  it feels weird to read "3730 unit-saucelabs tests passed".

To fix, here's my suggestion:
- Let `type` remain one of `unit` or `integration`. That way, the status will read "xx unit tests passed"
- In `postReport()`, infer three possible subtypes: `local-changes`, `saucelabs`, and `single-pass`.
- In the `postUrl`, post the status to `${reportBaseUrl}/${commitHash}/${type}/${subtype}/${action}`
- When the status is posted, they'll have names like `ampproject/tests/unit`, `ampproject/tests/unit/saucelabs`, `ampproject/tests/unit/local-changes`, `ampproject/tests/integration/single-pass`, etc.

I'd imagine this will need a small change in the bot code to parse `type` and `subtype`.